--- a/src/plugins/platforms/cocoa/qcocoamessagedialog.mm
+++ b/src/plugins/platforms/cocoa/qcocoamessagedialog.mm
@@ -87,11 +87,6 @@ bool QCocoaMessageDialog::show(Qt::WindowFlags windowFlags, Qt::WindowModality w
         return false;
     }
 
-    // Tahoe has issues with window-modal alert buttons not responding to mouse
-    if (windowModality == Qt::WindowModal
-        && QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSTahoe)
-        return false;
-
     // And without options we don't know what to show
     if (!options())
         return false;
