--- a/CMakeLists.txt	2024-12-29 21:34:43.219279282 +1000
+++ b/CMakeLists.txt	2024-12-29 21:34:59.495612349 +1000
@@ -3047,10 +3047,6 @@
 listtostr(EXTRA_CFLAGS _EXTRA_CFLAGS)
 set(EXTRA_CFLAGS ${_EXTRA_CFLAGS})
 
-if(USE_GCC OR USE_CLANG)
-  string(REGEX REPLACE "(^| )-I" "\\1 -isystem" EXTRA_CFLAGS "${EXTRA_CFLAGS}")
-endif()
-
 # Compat helpers for the configuration files
 
 if(EXISTS "${PROJECT_SOURCE_DIR}/VERSION.txt")
